stratum/hal/lib/bcm/bcm_chassis_manager_config_generator_sdk6.cc

```diff
@@ -0,0 +1,239 @@
+      // the port is flex or not. We dont use GetBcmChip as unit_to_bcm_chip_
+      // may not be populated when this function is called.
+      BcmChip::BcmChipType chip_type = BcmChip::UNKNOWN;
+      for (const auto& bcm_chip : target_bcm_chassis_map.bcm_chips()) {
+        if (bcm_chip.unit() == bcm_port.unit()) {
+          chip_type = bcm_chip.type();
+          break;
+        }
+      }
+      if (bcm_port.flex_port()) {
+        CHECK_RETURN_IF_FALSE(chip_type == BcmChip::TOMAHAWK ||
+                              chip_type == BcmChip::TRIDENT2)
+            << "Un-supported BCM chip type: "
+            << BcmChip::BcmChipType_Name(chip_type);
+        CHECK_RETURN_IF_FALSE(bcm_port.channel() >= 1 &&
+                              bcm_port.channel() <= 4)
+            << "Flex-port with no channel: " << bcm_port.ShortDebugString();
+        speed_bps =
+            flex_chip_to_channel_to_speed[chip_type][bcm_port.channel()];
+      } else {
+        speed_bps = bcm_port.speed_bps();
+      }
+    } else if (bcm_port.type() == BcmPort::MGMT) {
+      CHECK_RETURN_IF_FALSE(!bcm_port.flex_port())
+          << "Mgmt ports cannot be flex.";
+      speed_bps = bcm_port.speed_bps();
+    } else {
+      return MAKE_ERROR(ERR_INTERNAL)
+             << "Un-supported BCM port type: " << bcm_port.type() << " in "
+             << bcm_port.ShortDebugString();
+    }
+
+    // Port speed and diag port setting.
+    buffer << "portmap_" << bcm_port.logical_port() << "." << bcm_port.unit()
+           << "=" << bcm_port.physical_port() << ":"
+           << speed_bps / kBitsPerGigabit;
+    if (bcm_port.flex_port() && bcm_port.serdes_lane()) {
+      buffer << ":i";
+    }
+    buffer << std::endl;
+    buffer << "dport_map_port_" << bcm_port.logical_port() << "."
+           << bcm_port.unit() << "=" << bcm_port.diag_port() << std::endl;
+    // Lane remapping handling.
+    if (bcm_port.tx_lane_map() > 0) {
+      buffer << "xgxs_tx_lane_map_xe" << bcm_port.diag_port() << "."
+             << bcm_port.unit() << "=0x" << std::hex << std::uppercase
+             << bcm_port.tx_lane_map() << std::dec << std::nouppercase
+             << std::endl;
+    }
+    if (bcm_port.rx_lane_map() > 0) {
+      buffer << "xgxs_rx_lane_map_xe" << bcm_port.diag_port() << "."
+             << bcm_port.unit() << "=0x" << std::hex << std::uppercase
+             << bcm_port.rx_lane_map() << std::dec << std::nouppercase
+             << std::endl;
+    }
+    // XE ports polarity flip handling for RX and TX.
+    if (bcm_port.tx_polarity_flip() > 0) {
+      buffer << "phy_xaui_tx_polarity_flip_xe" << bcm_port.diag_port() << "."
+             << bcm_port.unit() << "=0x" << std::hex << std::uppercase
+             << bcm_port.tx_polarity_flip() << std::dec << std::nouppercase
+             << std::endl;
+    }
+    if (bcm_port.rx_polarity_flip() > 0) {
+      buffer << "phy_xaui_rx_polarity_flip_xe" << bcm_port.diag_port() << "."
```

Comment:
Let's not use the `phy_xaui_rx_polarity_flip_xe` format, but logical ports instead.